File: backends/java/zappy-domain/src/main/java/com/zappymart/domain/promotions/AppliedPromotion.java
package com.zappymart.domain.promotions;

import com.zappymart.domain.shared.Money;

import java.util.Objects;

public record AppliedPromotion(String code, PromotionKind kind, Money discount, boolean carriesShipping) {

    public AppliedPromotion {
        Objects.requireNonNull(code, "code");
        Objects.requireNonNull(kind, "kind");
        Objects.requireNonNull(discount, "discount");
    }
}

File: backends/kotlin/zappy-domain/src/main/kotlin/nl/zappymart/domain/promotions/AppliedPromotion.kt
package nl.zappymart.domain.promotions

import nl.zappymart.domain.shared.Money
import nl.zappymart.domain.shared.PromotionCode

data class AppliedPromotion(
    val code: PromotionCode,
    val kind: PromotionKind,
    val discount: Money,
) {
    val takesShippingAway: Boolean get() = kind == PromotionKind.FREE_SHIPPING
}
